src/packagedcode/build.py

```diff
@@ -9,9 +9,16 @@
 
 from collections import defaultdict
 import ast
+import io
+import json
 import logging
 import os
 
+from packageurl import PackageURL
+from pygments import highlight
+from pygments.formatter import Formatter
+from pygments.lexers import GroovyLexer
```

Comment:
interresting use of pygments :)

Reply:
@tardyp the idea of https://github.com/nexB/pygmars/ is to be used in tandem with pygments to create quickly some lightweight parsers for programming languages. 

This is used for instance to lex bash in https://github.com/nexB/scancode-toolkit/blob/develop/src/packagedcode/bashlex.py and parse it in https://github.com/nexB/scancode-toolkit/blob/develop/src/packagedcode/bashparse.py
In this case this is used to enable lightweight Bash parameter extension to parse the Apkbuild scripts from Alpine Linux (to collect metadata) in https://github.com/nexB/scancode-toolkit/blob/9ed2cb4e78a8b1b138bcb201f50203b43098febb/src/packagedcode/alpine.py#L161

The same approach will be used whenever there is a need to parse code to extract structured data as this is easier to read and maintain than a bunch of regex.  The  upcoming Yocto/Bitbake support likely can benefit from some of it too... and we should be able to parse also Autotools scripts that contain interesting metadata.

Reply:
Note that this is not perfect... and still only static. Next up will be to enable dependency resolution on the deps collected (which may require some live network access ... but this will be an option with proxy support for airgapped envt. ;) )